### ISSUE TITLE
PEP 723 support is faulty: uses wrong TOML field name

### ISSUE THREAD

**Issue by flying-sheep** [pypa/pipx#1187]
**Describe the bug**
#1100 implemented [PEP 723](https://peps.python.org/pep-0723/) wrong:

https://github.com/pypa/pipx/blob/5b18ddf5b2871749a1d9e90938da407c4ba3f7a4/src/pipx/commands/run.py#L351

this should be `"dependencies"`, not `"requirements"`.

**How to reproduce**

Create a PEP 723 compliant script:

```python
#!/usr/bin/env -S pipx run
# /// pyproject
# run.dependencies = ["httpx"]
# ///
import httpx
```

and run it using `pipx run ./script.py`. You’ll see

```pytb
Traceback (most recent call last):
  File "<string>", line 11, in <module>
ModuleNotFoundError: No module named 'requests'
```

**Expected behavior**

`pipx` should run it in a temporary environment.

Also the traceback should contain the file name, not `<string>`

**Comment by dukecat0** [comment]
Fixed in #1180.

**Comment by flying-sheep** [comment]
Ah, indeed. I searched the issues, but apparently not the PRs

**Comment by dukecat0** [comment]
> Also the traceback should contain the file name, not <string>

However, I think this issue still exists.

**Comment by maxwell-k** [comment]
The example in the issue report is unfortunately also out of date the most recent change to PEP 723 was to change:

```diff
-# /// pyproject
+# /// script
```

I don't want to write an example and risk getting it wrong myself, instead I will point to the example in the PEP
https://peps.python.org/pep-0723/#how-to-teach-this

For the background on this change please see https://discuss.python.org/t/how-would-you-like-to-declare-runtime-dependencies-and-python-requirements-for-pep-723/40418/70

**Comment by henryiii** [comment]
@maxwell-k this is also fixed in #1180.
